components/feature/qr/src/main/java/mozilla/components/feature/qr/QrFragment.kt

```diff
@@ -490,7 +509,17 @@ class QrFragment : Fragment() {
                     }
                 }
 
-                it.createCaptureSession(Arrays.asList(mImageSurface, surface), stateCallback, null)
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+                    val sessionConfig = SessionConfiguration(
+                        SessionConfiguration.SESSION_REGULAR,
+                        listOf(OutputConfiguration(mImageSurface as Surface), OutputConfiguration(surface)),
+                        backgroundExecutor as Executor,
+                        stateCallback
```

Comment:
It will be great if you could add some tests to exercise this path. Robolectric allows to configure the Android version to execute with the tests, you just have to add the `@Config(sdk = [Build.VERSION_CODES.P])` annotation above the test. For more details you can see http://robolectric.org/configuring/

Reply:
@Amejia481 How do I go about testing this? 
Here's what I came up with:
```
@Test
@Config(sdk = [Build.VERSION_CODES.P])
fun `a SessionConfiguration is used when creating preview session`() {
    val qrFragment = spy(QrFragment.newInstance(mock()))

    val imageReader: ImageReader = mock()
    whenever(imageReader.surface).thenReturn(mock())
    qrFragment.imageReader = imageReader

    val camera: CameraDevice = mock()
    qrFragment.cameraDevice = camera

    val textureView: AutoFitTextureView = mock()
    whenever(textureView.surfaceTexture).thenReturn(mock())
    qrFragment.textureView = textureView

    val previewRequestBuilder: CaptureRequest.Builder? = mock()
    whenever(camera.createCaptureRequest(anyInt())).thenReturn(previewRequestBuilder)

    qrFragment.previewSize = mock()
    qrFragment.createCameraPreviewSession()
    verify(camera).createCaptureSession(any<SessionConfiguration>())
}
```

I tried to mock `previewRequestBuilder` but I keep getting an error: 
```
Mockito cannot mock this class: class android.hardware.camera2.CaptureRequest$Builder
```

Reply:
@Amejia481 I'm having trouble testing this. Mockito doesn't seem to be able mock `CaptureRequest.Builder?` at: https://github.com/mozilla-mobile/android-components/blob/cd6b969eb21d5dc020b99cc11b15c404c058358a/components/feature/qr/src/main/java/mozilla/components/feature/qr/QrFragment.kt#L481-L485

I'm guessing because it's a static class.

This is what I have come up with:
```
        val qrFragment = spy(QrFragment.newInstance(mock()))

        val imageReader: ImageReader = mock()
        whenever(imageReader.surface).thenReturn(mock())
        qrFragment.imageReader = imageReader

        val camera: CameraDevice = mock()
        qrFragment.cameraDevice = camera

        val textureView: AutoFitTextureView = mock()
        whenever(textureView.surfaceTexture).thenReturn(mock())
        qrFragment.textureView = textureView

        val previewRequestBuilder: CaptureRequest.Builder? = mock()
        whenever(camera.createCaptureRequest(anyInt())).thenReturn(previewRequestBuilder)

        qrFragment.previewSize = mock()
        qrFragment.createCameraPreviewSession()
        verify(camera).createCaptureSession(any<SessionConfiguration>())
```